File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/keydistribution/KeyDistributor.java
package uk.nhs.nhsx.diagnosiskeydist.keydistribution;

import uk.nhs.nhsx.core.aws.s3.BucketName;
import uk.nhs.nhsx.core.aws.s3.ObjectKey;

import java.io.File;
import java.io.IOException;
import java.security.NoSuchAlgorithmException;

public interface KeyDistributor {

    void distribute(BucketName name, ObjectKey key, File binFile,
                    File sigFile) throws IOException, NoSuchAlgorithmException;

}

File: src/aws/lambdas/incremental_distribution/src/main/java/uk/nhs/nhsx/diagnosiskeydist/keydistribution/KeyDistributor.kt
package uk.nhs.nhsx.diagnosiskeydist.keydistribution

import uk.nhs.nhsx.core.aws.s3.BucketName
import uk.nhs.nhsx.core.aws.s3.ObjectKey
import java.io.File

fun interface KeyDistributor {

    fun distribute(
        name: BucketName,
        key: ObjectKey,
        binFile: File,
        sigFile: File
    )
}
